DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -200,6 +200,32 @@ private async ValueTask<TokenKind> GetTokenKindAsync(CancellationToken cancellat
         return Guild.Parse(response);
     }
 
+    public async IAsyncEnumerable<ThreadChannel> GetGuildChannelThreadsAsync(
+        string channelId,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        int currentOffset = 0;
+
+        while (true)
+        {
+            var url = new UrlBuilder()
+                .SetPath($"channels/{channelId}/threads/search")
+                .SetQueryParameter("offset", currentOffset.ToString())
+                .Build();
+
+            var response = await TryGetJsonResponseAsync(url, cancellationToken);
+
+            foreach(var threadJson in response?.GetProperty("threads").EnumerateArray() ?? Enumerable.Empty<JsonElement>())
+                yield return ThreadChannel.Parse(threadJson);
+
+            if (!(response?.GetProperty("has_more").GetBoolean() ?? false))
+            {
+                break;
+            }
```

Comment:
Can probably simplify further?

```suggestion
            var response = await TryGetJsonResponseAsync(url, cancellationToken);
            if (response is null)
            		break;

            foreach (var threadJson in response.GetProperty("threads").EnumerateArray())
                yield return ThreadChannel.Parse(threadJson);

            if (!response.GetProperty("has_more").GetBoolean())
                break;
```

Reply:
Wasn't able to get this one to compile due to this error:
`CS1061 'JsonElement?' does not contain a definition for 'GetProperty' and no accessible extension method 'GetProperty' accepting a first argument of type 'JsonElement?' could be found (are you missing a using directive or an assembly reference?)`

I got this error before, which I solved by using null-conditional and null-coalescing operators when parsing `response`. The same solution seems to work here.
Might be doing something wrong on my end? Your solution certainly seems to make more sense.

Reply:
Ah, it has to be `response.Value.GetProperty("has_more")` because it's a value type

Reply:
Yep, you're right! That worked great.

Resolved in d88bbc8.